design/baremetal-operator/firmware-interface.md

```diff
@@ -0,0 +1,221 @@
+<!--
+ This work is licensed under a Creative Commons Attribution 3.0
+ Unported License.
+
+ http://creativecommons.org/licenses/by/3.0/legalcode
+-->
+
+# firmware-interface
+
+## Status
+
+implementable
+
+## Summary
+
+This design document describes a Metal3 API for using the Firmware Interface
+feature added to Ironic, this will allow updates of firmware of bmc and bios
+in BMH that are compatible with Redfish.
+
+## Motivation
+
+When provisioning new machines to clusters, it may be necessary to upgrade
+or downgrade the servers BIOS and firmware to specific versions to ensure
+the server is configured as it is in their validated pattern.
+
+Redfish has a common API that allow users to execute a firmware update, the
+[proposal](https://review.opendev.org/c/openstack/ironic-specs/+/878505) was
+accepted in the Ironic community, and is already implemented.
+
+### Goals
+
+This proposal does not attempt to fulfill the following goals:
+
+- Allow Firmware updates for the BMC and BIOS in hardware that compatible
+  with Redfish.
+
+### Non-Goals
+
+- Update Firmware using non Redfish means
+- Update firmware that are not for BMC and BIOS
+
+## Proposal
+
+### User Stories
+
+#### Story 1
+
+As an operator I want to install specific versions of firmware in my
+machines before installing the Operating System.
+
+## Design Details
+
+This proposes a new Custom Resource Definition (CRD) to store the information
+about the firmware components from Ironic. The initial version stores only
+the firmware information about BMC and BIOS, we may expand this to other
+components in the future.
+The new CRD is named `HostFirmwareComponents` and will consist of the
+following:
+
+- `componentsData` - the current Firmware Components and it information,
+  retrieved from Ironic via the [Firmware API](https://docs.openstack.org/api-ref/baremetal/)
+  will be stored in the `Status` section.
+- `updateComponents` - Firmware components to be updated via Ironic will be
```

Comment:
We try to avoid imperative language in the declarative API. Maybe just call these "updates"?